lib/formContext.ts

```diff
@@ -147,40 +147,163 @@ export const getElementById = (elements: FormElement[], id: string) => {
 };
 
 /**
- * Checks a page/group visibility based on the current form values groupHistory.
+ * Returns values array but with matched ids for checkboxes and radios.
+ *
+ * @param formElements
+ * @param values
+ * @returns {FormValues} - Returns a new FormValues object with matched ids for checkboxes and radios.
+ */
+export const getValuesWithMatchedIds = (formElements: FormElement[], values: FormValues) => {
+  const newValues = { ...values };
+  Object.entries(values).forEach(([key, value]) => {
+    if (key !== "currentGroup" && key !== "groupHistory" && key !== "matchedIds") {
+      const el = getElementById(formElements, key);
+
+      if (value && el && (el.type === "checkbox" || el.type === "radio")) {
```

Comment:
We have a function called 

```
export const findChoiceByValue = (
```

We don't have to use it but might make sense --- just pointing it out.